joint_trajectory_controller/src/joint_trajectory_controller.cpp

```diff
@@ -450,9 +453,10 @@ bool JointTrajectoryController::read_state_from_command_interfaces(JointTrajecto
   auto assign_point_from_interface =
     [&](std::vector<double> & trajectory_point_interface, const auto & joint_interface)
   {
-    for (size_t index = 0; index < dof_; ++index)
+    for (size_t index = 0; index < num_cmd_joints_; ++index)
     {
-      trajectory_point_interface[index] = joint_interface[index].get().get_value();
+      trajectory_point_interface[map_cmd_to_joints_[index]] =
+        joint_interface[index].get().get_value();
```

Comment:
```suggestion
        joint_interface[map_cmd_to_joints_[index]].get().get_value();
```
As it is reading from the state, shouldn't it be using the map index?

Reply:
Probably this makes sense as it is right now

Reply:
it is called with `assign_point_from_interface(state.positions, joint_command_interface_[0]);`
`state` has size of `dof`, `joint_command_interface_` only of the actual command joints

Reply:
Got it. Thanks